### ISSUE TITLE
[BUG]  - Mailbox component - Inability to load messages in threads while using own data

### ISSUE THREAD

**Issue by tudorbertiean** [nylas/components#309]
**Describe the bug**
When using the mailbox component with my own data (structured as threads), I am able to open threads but not individual messages. The last message which attempts to load and show by default gets stuck in an infinite loading state, and then I receive the following error message when I attempt to open another individual message in the thread:
<img width="1031" alt="Screen Shot 2022-01-04 at 10 26 13 PM" src="https://user-images.githubusercontent.com/11579557/148156304-f41099b6-5a29-4c3f-8b5f-3adeadda60d1.png">

Error 1: Last message stuck in infinite loading state
Error 2: Attempting to open any other message gives me the above error

**To Reproduce**
Mailbox settings gives access to all Nylas users, not individuals.

Component Loaded:
```
<nylas-mailbox id={emailId} access_token={accessToken}> </nylas-mailbox>

```

Data Loaded:
```
const mailbox = document.getElementsByTagName("nylas-mailbox")[0];
mailbox.all_threads = allThreads;

```

Threads (allThreads):
Retrieved from a get request to Nylas' API via https://api.nylas.com/threads/{threadId}?view=expanded

**Expected behavior**
Clicking on messages in threads should not error out or load infinitly.

**Version:**
1.1.5

**Comment by AaronDDM** [comment]
Hey @tudorbertiean,

Thanks for reporting this issue. We're investigating this and will update with more information as soon as we can.

**Comment by AaronDDM** [comment]
Hey @tudorbertiean ,

So looking into this issue, the mailbox component will not attempt to make a fetch request for thread message bodies if the thread information is passed in via the `all_threads` prop. So you must include the thread message bodies yourself.

There are however some issues:
1. There should not be an error (as per your screenshot)
2. There was no `messageClicked` dispatch event that could have been used to hydrate the thread message body

Both of the above two issues have been resolved in #315.

So for your case, you'll have to do either:
(a) pass in `all_threads` with full message bodies
(b) use the `threadClicked` and `messageClicked` dispatched events to identify the thread & message clicked and hydrate them accordingly with the message bodies.

For example:
```js
        mailbox.addEventListener("threadClicked", (v) => {
          let { thread } = v.detail;
          thread = {
            ...thread,
            messages: thread.messages.map((m) => {
              m.body = "THIS IS A TEST";
              return m;
            })
          }
        });
```

This will make it so that each message will have the body "THIS IS A TEST".

**Comment by tudorbertiean** [comment]
Great, thanks so much for the quick responses and turnaround @AaronDDM!
